pandas/core/internals/blocks.py

```diff
@@ -1950,6 +1955,35 @@ def shift(self, periods, axis=0):
     def _ftype(self):
         return getattr(self.values, '_pandas_ftype', Block._ftype)
 
+    def _unstack(self, unstacker_func, new_columns):
```

Comment:
Ugh, so this is bad. There are several issues.

1. `_unstack_frame` partially applies `_Unstacker` and then applies it to the to the *blocks*` of a frame via `BlockManger._unstack`. For `Series[EA]`, we preempt the `_Unstacker` down in https://github.com/pandas-dev/pandas/pull/23284/files#diff-fef81b7e498e469973b2da18d19ff6f3R403.
2. modifying a partially applied function is probably a bad idea, but if I want to use a similar strategy, I need to extract the `fill_value` and then update it to use `-1`.
3. I do a dummy unstack on an empty array just to get the index length. This could maybe be avoided
4. After that, this is the same as NonConsolidatableMixin._untsack.

So consider this a WIP for now.

Reply:
Maybe you can change the `Block.unstack` signature to not pass through the partial object, but the actual object and keywords? So it less hacky to change something?

Reply:
At a glance, it looks like the `_Unstacker.__init__` is somewhat expensive. I suspect this was done originally for performance.

Reply:
Ah, but I could at least pass through the kwargs in addition to the partially applied _Unstacker.

Reply:
That would let us avoid the hacks with the `functools.partial`.

Reply:
But is the partial doing any caching? I would expect the `__init__` not yet been run in the partial, and after that it is called again for each block it seems? 
(but might be wrong, didn't look in detail)

Reply:
Right, Your correct.